src/main/scala/esmeta/analyzer/domain/state/TypeDomain.scala

```diff
@@ -324,14 +333,19 @@ object TypeDomain extends state.Domain {
               else res ||= nts(propIdx).fold(AbsentT)(AstT(_))
             }
           case Inf => res ||= AstT
-        prop.str match
-          case Fin(ss) =>
-            for (s <- ss) s match
-              case "parent" => res ||= AstT
-              case name =>
-                if (cfg.grammar.nameMap contains name) res ||= AstT(name)
-                else () // TODO warning(s"invalid access: $name of $ast")
+        handlePropStr(prop)
+      case AstNameTy(names) =>
+        prop.math match
+          case Fin(ns) =>
+            for {
+              n <- ns
+              if n.isValidInt
+              propIdx = n.toInt
+            } {
+              res ||= AstT
+            }
```

Comment:
Please simplify this code.

Reply:
I tried to solve the comment with https://github.com/es-meta/esmeta/pull/157/commits/1649af9e20f5e41d9a7461c9bb54e3839976f905